api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/BaseDao.java

```diff
@@ -563,58 +563,61 @@ protected boolean containsAlias(Iterator<CriteriaImpl.Subcriteria> subcriteriaIt
 	}
 	
 	protected void handleParticipantReference(Criteria criteria, ReferenceAndListParam participantReference) {
-		if (participantReference != null && lacksAlias(criteria, "ep")) {
-			criteria.createAlias("encounterProviders", "ep");
+		if (participantReference != null) {
+			if (!lacksAlias(criteria, "en")) {
+				criteria.createAlias("en.encounterProviders", "ep");
+			}
```

Comment:
This doesn't look right to me, since it seems like it could try to create the `ep` alias twice. The reason for the `lacksAlias` guard is to ensure that we only bind an alias once. I would refactor this to something like this:

```
if (lackAlias(criteria, "en")) {
	return;
}

if (lacksAlias(criteria, "ep")) {
	criteria.createAlias("en.encounterProviders", "ep");
}

// etc.
```